messages/src/lib.rs

```diff
@@ -216,25 +276,83 @@ impl Decode for Duration {
 
 impl Display for Duration {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{} seconds", self.0)
+        write!(f, "{} time_precision units", self.0)
     }
 }
 
-/// DAP protocol message representing an instant in time with a resolution of seconds.
+/// DAP protocol message representing an instant in time as a multiple of the task's time
+/// precision. The value represents the number of time_precision intervals since the Unix epoch
+/// (January 1st, 1970, at 0:00:00 UTC).
+///
+/// To convert between this representation and real-world timestamps (seconds since epoch),
+/// use the conversion methods that take a [`TimePrecision`] parameter.
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
 pub struct Time(u64);
 
 impl Time {
     /// Construct a [`Time`] representing the instant that is a given number of seconds after
     /// January 1st, 1970, at 0:00:00 UTC (i.e., the instant with the Unix timestamp of
-    /// `timestamp`).
-    pub const fn from_seconds_since_epoch(timestamp: u64) -> Self {
-        Self(timestamp)
+    /// `timestamp`), given the task's time precision.
+    ///
+    /// The timestamp will be rounded down to the nearest multiple of time_precision.
+    ///
+    /// # Arguments
+    /// * `timestamp` - Unix timestamp in seconds
+    /// * `time_precision` - The task's time precision
+    ///
+    /// # Panics
+    /// Panics if `time_precision.as_seconds()` is 0.
+    ///
+    pub fn from_seconds_since_epoch(timestamp: u64, time_precision: &TimePrecision) -> Self {
+        let precision_secs = time_precision.as_seconds();
+        assert!(precision_secs > 0);
+        Self(timestamp / precision_secs)
+    }
+
+    /// Construct a [`Time`] representing the provided [`SystemTime`], given the task's
+    /// time precision.
+    ///
+    /// The time will be rounded down to the nearest multiple of time_precision.
+    ///
+    /// # Arguments
+    /// * `time` - SystemTime
+    /// * `time_precision` - The task's time precision
+    ///
+    /// # Panics
+    /// Panics if `time_precision.as_seconds()` is 0, or `time` is before the UNIX Epoch.
+    ///
+    pub fn from_systemtime(time: SystemTime, time_precision: &TimePrecision) -> Self {
```

Comment:
I'm not sure we should have this method. `SystemTime` is just one of the many representations of time that a DAP user could be working with, and all we're really doing here is saving them the trouble of computing seconds since the epoch. It looks like the only place we use it is in an example in the client crate. I think we should avoid committing to this public API until we have a stronger case for maintaining it.

Reply:
Good idea. Removed.